### DISCUSSION TITLE
is possible to server from a file imported using include_bytes! macro?

### DISCUSSION THREAD

**Discussion by johnyenter-briars** [rwf2/Rocket#1929] [Questions] [answered]
Greetings Rocket Experts,

I would like to package my Rocket application as a stand alone binary. 

So, I would like to use `include_bytes!` macro to include a singular html file statically. 

Is there a way I can still serve the file using `NamedFile` or something similar? I looked at the [docs](https://api.rocket.rs/v0.5-rc/rocket/fs/struct.NamedFile.html) and didn't see any applicable API.

Am I missing a possible solution? Or does this feature just not exist in Rocket?

Thanks!

**Comment by jebrosen** [comment, marked as answer]
`NamedFile` is used specifically to serve files from disk; there's no need to use it at the same time as `include_bytes!`.

> Am I missing a possible solution? Or does this feature just not exist in Rocket?

Not only is it possible, it's intended to be pretty straightforward to do things like this! For example:

```rust
use rocket::http::ContentType;

#[get("/example.png")]
fn image() -> (ContentType, &'static [u8]) {
    (ContentType::PNG, include_bytes!("/tmp/example.png"))
}
```

**Reply by johnyenter-briars** [reply]
Ok - thank you! Will NamedFile still package the bytes into the stand along binary when built for release? The objective is that I don't want to serve files from disk, I want to serve files which are packaged in the stand alone binary.
